x-pack/plugins/elastic_assistant/server/lib/langchain/executors/types.ts

```diff
@@ -14,14 +14,21 @@ import type { LangChainTracer } from 'langchain/callbacks';
 import { RequestBody, ResponseBody } from '../types';
 
 export interface AgentExecutorParams {
+  alertsIndexPattern?: string;
   actions: ActionsPluginStart;
+  allow?: string[];
+  allowReplacement?: string[];
+  assistantLangChain: boolean;
   connectorId: string;
   esClient: ElasticsearchClient;
   kbResource: string | undefined;
   langChainMessages: BaseMessage[];
   llmType?: string;
   logger: Logger;
+  onNewReplacements: (newReplacements: Record<string, string>) => void;
```

Comment:
nit below to make optional: https://github.com/elastic/kibana/pull/172542/files#r1416185132